core/src/main/java/me/lokka30/treasury/plugin/core/utils/PluginVersion.java

```diff
@@ -12,25 +12,33 @@
 public final class PluginVersion {
 
     private final short majorRevision, minorRevision, patchRevision;
-    private final boolean developmentVersion;
-    private final String stringVersion;
+    private final boolean developmentVersion, release;
+    private final String commit, stringVersion;
 
     public PluginVersion(@NotNull String input, @NotNull Logger logger) {
         Objects.requireNonNull(input, "input");
         Objects.requireNonNull(logger, "logger");
         this.stringVersion = input;
-        if (input.endsWith("-SNAPSHOT")) {
-            developmentVersion = true;
-            input = input.replace("-SNAPSHOT", "");
-        } else {
-            developmentVersion = false;
+
+        if (input.indexOf('.') == -1 || input.indexOf('-') == -1) {
+            throw new IllegalArgumentException(
+                    "Your current Treasury version is not formatted correctly! Please inform a Treasury maintainer as soon as possible.");
         }
-        if (input.indexOf('.') == -1) {
-            throw new IllegalArgumentException("Illegal Treasury version found! " + "If this is a live environment version or you're a developer doing changes, please fix this ASAP!");
+        // parse commit before version
+        String[] bigParts = input.split("-");
+        if (bigParts.length != 3) {
+            throw new IllegalArgumentException(
+                    "Your current Treasury version is not formatted correctly! Please inform a Treasury maintainer as soon as possible.");
         }
-        String[] parts = input.split("\\.");
+        this.commit = bigParts[1];
+
+        this.developmentVersion = bigParts[2].equalsIgnoreCase("SNAPSHOT");
+        this.release = bigParts[2].equalsIgnoreCase("RELEASE");
+
+        String[] parts = bigParts[0].split("\\.");
         if (parts.length != 3) {
-            throw new IllegalArgumentException("Illegal Treasury version found! " + "If this is a live environment version or you're a developer doing changes, please fix this ASAP!");
+            throw new IllegalArgumentException(
+                    "Your current Treasury version is not formatted correctly! Please inform a Treasury maintainer as soon as possible.");
```

Comment:
Instead of all these checks and throws we could use regex to check if the version format is valid

Reply:
Regex is something I don't want to get involved with. If you give me the pattern and the code to handle it, sure I'd use it. But I am not implementing such thing at all.

Reply:
I'm not very familiar with regex but this should do it
```Java
private static final Pattern VERSION_PATTERN = Pattern.compile("\\d+\\.\\d+\\.\\d+-[0-9A-Fa-f]{7}-(SNAPSHOT|RELEASE)");
...
if(!VERSION_PATTERN.matcher(input).matches()) {
  throw new IllegalArgumentException("Your current Treasury version is not formatted correctly! Please inform a Treasury maintainer as soon as possible."):
}
```
The pattern searches this:
`\d+\.`: Any decimal character (1 or more) followed by a '.'
`\d+\.`: Any decimal character (1 or more) followed by a '.'
`\d+-`: Any decimal character (1 or more) followed by a '-'
`[0-9A-Fa-f]{7}-`: Seven hexadecimal characters followed by a '-'
`(SNAPSHOT|RELEASE)`. Literal string "SNAPSHOT" or literal string "RELEASE"

Reply:
The online tester didn't work, however I've tested the following code:

```java
  public static void main(String[] args) {
    Pattern VERSION_PATTERN = Pattern.compile("\\d+\\.\\d+\\.\\d+-[0-9A-Fa-f]{7}-(SNAPSHOT|RELEASE)");
    String matchAgainst = "1.0.0-003b4a0-SNAPSHOT";

    Matcher matcher = VERSION_PATTERN.matcher(matchAgainst);
    if (!matcher.matches()) {
      System.err.println("Nope.");
      return;
    }

    System.out.println("Match successful");
  }
```

And it looks like it works: 
![img](https://img.mrivanplays.com/ciukfwxkbm.png)

I'm implementing it.